.github/workflows/copywrite.yml

```diff
@@ -0,0 +1,19 @@
+name: Check Copywrite Headers
+
+on:
+  push: {}
+
+jobs:
+  copywrite:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - uses: hashicorp/setup-copywrite@867a1a2a064a0626db322392806428f7dc59cb3e # v1.1.2
+        name: Setup Copywrite
+        with:
+          version: v0.16.4
+          archive-checksum: c299f830e6eef7e126a3c6ef99ac6f43a3c132d830c769e0d36fa347fa1af254
+      - name: Check Header Compliance
+        run: copywrite headers --plan
+permissions:
+  contents: read
```

Comment:
Though I don't know the full extent of what the `copywrite` command does, I imagine it should be fine.  The contents of the repo itself are pulled by `actions/checkout` in the first step, which only needs this `contents:read` here.


Beyond that, according to the [copyright repo's readme](https://github.com/hashicorp/copywrite#github-authentication), for some operations

> A GITHUB_TOKEN environment variable can be used with a Personal Access Token

I suspect the GITHUB_TOKEN (auto-provided by GHA), if used at all in our case, will mainly be to uniquely identify the caller so our private runners don't get some egress-IP rate-limited or something by GH API.